server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/model/MockSMTPBehaviorInformation.java

```diff
@@ -0,0 +1,165 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mock.smtp.server.model;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import com.google.common.base.Preconditions;
+
+public class MockSMTPBehaviorInformation {
+
+    interface RemainingAnswersCounter {
+
+        void decrease();
+
+        boolean hasRemainingAnswers();
+
+        Optional<Integer> getValue();
```

Comment:
I don't like this "testing purpose" method being part of the interface as this makes little sens for the "unlimited case".

Can't we find a way of writing these tests without going too deep in implementation knowledge but but staying at the API level?

Reply:
It's not only for testing but I need it for seralization, where the MockSmtpBehaviorInformation can provide number of remaining answers to the http client

Reply:
I feel it ugly too, but IDK what can I do, I tried to put a logic to resloving counting in a dedicated method of MockSMTPBehaviorInformation class, the big class even looks more ugly